cpp/src/jit/parser.h

```diff
@@ -34,12 +37,80 @@ will have.
 @return The output CUDA `__device__` `__inline__` function
 */
 
-std::string parse_single_function_ptx(const std::string& src,
+namespace cudf {
+namespace jit {
+
+class ptx_parser {
+
+private:
+  
+  std::string ptx;
+
+  std::string function_name;
+
+  std::string output_arg_type;
+
+  std::set<int> pointer_arg_list;
+
+  std::map<std::string, std::string> input_arg_list;
+
+private:
+  
+  std::string parse_function_header(const std::string& src);
```

Comment:
Docs for each of these. I think every function in this file should have a brief comment showing an example input string and output string to make clear what it does.